pkg/miner/endorsementpool/endorsement_pool.go

```diff
@@ -0,0 +1,194 @@
+package endorsementpool
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/wavesplatform/gowaves/pkg/crypto/bls"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+)
+
+type key struct {
+	blockID       proto.BlockID
+	endorserIndex int32
+}
+
+func makeKey(blockID proto.BlockID, idx int32) key {
+	return key{blockID: blockID, endorserIndex: idx}
+}
+
+type heapItemEndorsement struct {
+	eb         *proto.EndorseBlock
+	endorserPK bls.PublicKey
+	balance    uint64
+	seq        uint64
+}
+type EndorsementPool struct {
+	mu        sync.Mutex
+	seq       uint64
+	byKey     map[key]*heapItemEndorsement
+	items     []*heapItemEndorsement // always sorted
+	conflicts []proto.EndorseBlock
+}
+
+func NewEndorsementPool() *EndorsementPool {
+	return &EndorsementPool{
+		byKey: make(map[key]*heapItemEndorsement),
+	}
+}
```

Comment:
Does not see `limit` in endorsements. As I said it should be taken from `FunctionalitySettings`.